### PR TITLE
Replace jQuery slider with pure React

### PR DESCRIPTION
**Pull request by sfentress**
@knowuh Could you take a look at this?

I was hoping that we'd be able to get rid of the jQuery dependency entirely, but it looks like we're still using it in a number of places. Maybe one day we'll get an intern to go through all the references armed with http://youmightnotneedjquery.com/

### CODE REVIEW

**Comment by knowuh** [comment]
This looks much improved.  Thanks @sfentress !

**Comment by sfentress** [comment]
I will add a fix to solve the issue where the handle goes outside the range if it's set to an out-of-bounds value.

File: src/code/views/value-slider-view.coffee [hunk 1]
```diff
@@ -125,34 +111,95 @@ ValueSlider = React.createClass
...
+    e.preventDefault()
+
+  handleStart: (e) ->
+    @handleNoop(e)
+    document.addEventListener 'mousemove', @handleDrag
+    document.addEventListener 'mouseup',   @handleEnd
+
+  handleEnd: ->
+    document.removeEventListener 'mousemove', this.handleDrag
+    document.removeEventListener 'mouseup',   this.handleEnd
```
Comment (sfentress) [inline src/code/views/value-slider-view.coffee]: No reason at all. I'll clean that up with my other changes.

**Comment by sfentress** [comment]
@knowuh I've updated this to to fix the two bugs, slider going out of bounds and slider dragging a selected node.

https://www.pivotaltracker.com/story/show/134443309
https://www.pivotaltracker.com/story/show/134443393

**Comment by knowuh** [comment]
This looks fine @sfentress -- thanks

**Pull request merged**